lumicks/pylake/kymotracker/kymotracker.py

```diff
@@ -486,6 +509,9 @@ def refine_tracks_gaussian(
         When supplied, the background is not estimated but fixed at this value.
     """
     assert overlap_strategy in ("ignore", "skip", "multiple")
+    assert isinstance(window, int) or np.issubdtype(
+        window.dtype, np.integer
+    ), "window parameter should be integer"
```

Comment:
assertions in code that is not testing code should be avoided, and a proper exception should be raised instead

> An assert’s primary function should be to isolate a bug’s root cause and should never be used for standard error handling. While asserts can be beneficial, it’s essential to use them for debugging and tests, and not to control the flow of the logic for the program or application and use them only in safe situations.

nice readings:

https://towardsdatascience.com/2-caveats-to-avoid-when-using-python-assert-statement-8060e123c26a
https://snyk.io/blog/the-dangers-of-assert-in-python/

Reply:
I guess we might need a ticket on pylake to clean all assertions up...

Reply:
Huh. TIL. Good to know, thank you @alessiamarcolini .

> I guess we might need a ticket on pylake to clean all assertions up...

Yeah, I was just thinking that as well. I'll pick this up this sprint. Shouldn't take too long.

Reply:
So, I was looking at this again. I will make a separate PR for fixing these asserts up in other places. I also dropped the int check I added here. I've made a ticket to look at this particular case in a bit more detail, since the docstring marks it as an int (which is why I added the assert), but from what I can tell, downstream can in principle handle it being float.